src/utils.ts

```diff
@@ -265,6 +265,38 @@ module powerbi.extensibility.visual.utils {
             return GranularityNames[index].granularityType;
         }
 
+        public static getGranularityPropsByMarker(marker: string): GranularityName {
+            const index: number = Utils.findIndex(GranularityNames, (granularity: GranularityName) => {
+                return granularity.marker === marker;
+            });
+
+            return GranularityNames[index];
+        }
+
+        /**
+         * Returns the granularity type of the given granularity name
+         * @param granularityName The name of the granularity
+         */
+        public static getGranularityTypeByMarker(granularityMarker: string): GranularityType {
+            const index: number = Utils.findIndex(GranularityNames, (granularity: GranularityName) => {
```

Comment:
Code duplication. To get an `index` you should call `getGranularityPropsByMarker`